hal/armv7m/stm32/l4/console.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Phoenix-RTOS
+ *
+ * plo - operating system loader
+ *
+ * Console
+ *
+ * Copyright 2021 Phoenix Systems
+ * Authors: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+#include "stm32l4.h"
+
+
+static struct {
+	volatile u32 *base;
+	unsigned cpufreq;
+} halconsole_common;
+
+
+enum { cr1 = 0, cr2, cr3, brr, gtpr, rtor, rqr, isr, icr, rdr, tdr };
+
+
+void hal_consolePrint(const char *s)
+{
+	while (*s) {
+		if (~(*(halconsole_common.base + isr)) & 0x80)
+			continue;
+
+		*(halconsole_common.base + tdr) = *(s++);
+	}
+
+	while (~(*(halconsole_common.base + isr)) & 0x80);
+
+	return;
+}
+
+
+void console_init(void)
+{
+	struct {
+		void *base;
+		u8 uart;
+	} uarts[] = {
+		{ (void *)0x40013800, pctl_usart1 }, /* USART1 */
+		{ (void *)0x40004400, pctl_usart2 }, /* USART2 */
+		{ (void *)0x40004800, pctl_usart3 }, /* USART3 */
+		{ (void *)0x40004c00, pctl_uart4 }, /* UART4 */
+		{ (void *)0x40005000, pctl_uart5 }  /* UART5 */
+	};
+
+	const int uart = 1, port = pctl_gpiod, txpin = 5, rxpin = 6, af = 7;
```

Comment:
What is the reason to provide `uarts[]` if the `port`, `txpin` etc. are hardcoded?

Reply:
Copy-paste from kernel, will fix it in both places in the future

Reply:
Ok